Assets/PackageStagingArea/eu.netherlands3d.cameras/Runtime/Scripts/Cameras/Compass.cs

```diff
@@ -0,0 +1,130 @@
+{
+    public class Compass : MonoBehaviour
+    {
+        public RectTransform arrowTransform;
+        public Color NorthColor;
+        
+        private Camera cameraComponent;
+        private FreeCamera freeCamera;
+        private Transform cameraTransform;
+        private bool cancelingAnimation = false;
+        private Image arrowImage;
+        private Color arrowColor;
+
+        private const float directionMargin = 0.1f;
+        private const float northAngleMargin = 1.0f;
+
+        [Header("Animation tweaks")]
+        [SerializeField] private float snapSpeed = 10f;
+
+        [Header("Input")][SerializeField] private InputActionAsset inputActionAsset;
+        private InputAction rotateAction;
+        private InputActionMap cameraActionMap;
+
+        private void Awake()
+        {
+            cameraComponent = GetComponent<Camera>();
+            freeCamera = GetComponent<FreeCamera>();
+            cameraTransform = cameraComponent.transform;
+
+            arrowImage = arrowTransform.GetComponent<Image>();
+            arrowColor = arrowImage.color;
+
+            //when user rotates, cancel the animation if active
+            cameraActionMap = inputActionAsset.FindActionMap("Camera");
+            rotateAction = cameraActionMap.FindAction("RotateModifier");
+            rotateAction.started += CancelAnimation;
+        }
+
+        private void Update()
+        {
+            arrowTransform.SetRotationZ(cameraTransform.transform.eulerAngles.y);
+            float angle;
+            if(cameraComponent.orthographic)
+            {               
+                angle = Vector3.SignedAngle(cameraTransform.up, Vector3.forward, Vector3.up);                
+            }
+            else
+            {                                
+                angle = Mathf.Rad2Deg * Mathf.Atan2(cameraTransform.forward.z, cameraTransform.forward.x) - 90;                
+            }            
+            if(Mathf.Abs(angle) < northAngleMargin)
+                arrowImage.color = NorthColor;
+            else
+                arrowImage.color = arrowColor;
+        }
+
+        public void SwitchToNorth()
+        {
+            cancelingAnimation = false;
+
+            // Find the tile handlers that are currently active, to disable them at the start of the animation
+            // and reactivate them after the animation to prevent glitching from LOD switches
+            var activeTileHandlers = FindObjectsByType<CartesianTiles.TileHandler>(
+                FindObjectsInactive.Exclude,
+                FindObjectsSortMode.None
+            );
+
+            //set target rotation
+            Quaternion rotateTo = Quaternion.LookRotation(Vector3.forward, Vector3.up);
+            rotateTo.eulerAngles = new Vector3(cameraTransform.eulerAngles.x, rotateTo.eulerAngles.y, cameraTransform.eulerAngles.z);
+
+            StartCoroutine(CameraAnimationLoop(rotateTo, cameraTransform,
```

Comment:
Dit kan je ook met DOTween oplossen, we hebben bewust een tweening library in het project voor animaties en sequences